mypy/checker.py

```diff
@@ -2955,10 +2955,14 @@ def check_metaclass_compatibility(self, typ: TypeInfo) -> None:
         if typ.metaclass_type is None and any(
             base.type.metaclass_type is not None for base in typ.bases
         ):
+            explanation = typ.explain_metaclass_conflict() or ""
+            if explanation:
+                explanation = f" - {explanation}"
             self.fail(
                 "Metaclass conflict: the metaclass of a derived class must be "
-                "a (non-strict) subclass of the metaclasses of all its bases",
+                f"a (non-strict) subclass of the metaclasses of all its bases{explanation}",
```

Comment:
I'm aware that adding the explanation makes error message differ from the runtime error. However, since type hints sometimes don't follow the actual runtime metaclasses (s. issue this PR is supposed fix), I think this information can be valuable.

Reply:
Mypy errors don't necessarily have to mirror runtime errors exactly. It's nice to be consistent, but maybe we have more information so we can give a more detailed error, or we think a different wording is more informative.